device-connectors/src/testflinger_device_connectors/devices/muxpi/muxpi.py

```diff
@@ -86,6 +86,38 @@ def _run_control(self, cmd, timeout=60):
             raise ProvisioningError(e.output)
         return output
 
+    def check_control_alive(self):
+        """Check if the control host is alive"""
+        try:
+            self._run_control("true")
+        except (ProvisioningError, subprocess.SubprocessError):
+            raise ProvisioningError(
+                "Control host is not responding, provisioning can't proceed!"
+            )
+
+    def reboot_control_host(self):
+        """
+        Reboot the control host
+
+        Sometimes the control host can end up in an unstable condition which
+        isn't easy to detect until it's too late. Since nothing else should
+        be using the control host, reboot it before provisioning to ensure
+        it's in a known good state.
+        """
+        self.check_control_alive()
+        logger.info("Rebooting control host")
+        # Use shutdown instead of reboot to avoid ssh disconnecting too quickly
+        self._run_control("sudo shutdown -r +1")
```

Comment:
This with the sleep(60) introduces a perfect race :)
There's a small, but possible chance of python resuming before the shutdown machine completed, yielding a premature completion of `check_control_alive()`

Reply:
Oh, and fixing this may render the last check not needed.

Reply:
Good points, thanks. The changes I just pushed should take care of both I think